### ISSUE TITLE
CUE support

### ISSUE THREAD

**Issue by okukharchuk** [owncloud/music#709]
It would be really nice to have CUE support.

**Comment by okukharchuk** [comment]
It should be pretty straightforward on backend side. I think just add something like "offset" into Track. Cue parsing already supported by getID3. I could probably do it on backend and in API but I'm not really familiar with JS and frontend dev.

**Comment by paulijar** [comment]
I'm not familiar with the CUE files, but based on what I have read, apparently the idea is that you have a music album ripped as a single MP3 file (or any other audio format), and then the CUE file describes the offsets, lengths, and metadata of each contained track. Is this what you had in mind?

What was your idea about presenting this to the end user? Should it be completely transparent so that the user wouldn't see any difference in browsing and playing tracks, regardless of if the tracks are stored as individual MP3 files or as part of some larger bundle described by a CUE file?

**Comment by coamtaker618** [comment]
ooohhh, this would be amazing!

Your summary is correct @paulijar   instead of splitting a CD into individual files, you get one audio file (mp3, wav, ogg, whatever) that's pure audio and a cue text file that has all the metadata (start/stop time of tracks, artist/album info, ect).

I haven't looked at many cue files, so I don't know what options are available (maybe there's more info in the cue file to give you even more options?) but I would be happy presenting audio + cue as you say, a transparent folder that looks like a file a track :)

**Comment by ei8fdb** [comment]
> you get one audio file (mp3, wav, ogg, whatever) that's pure audio and a cue text file that has all the metadata (start/stop time of tracks, artist/album info, ect).

@coatmaker618 I'm not familiar with CUE files either. What would a user see in the music file folder? Something like this?

Album name
- album_name.mp3
- album_name.cue

Or is the cue file "embedded" in the audio file?

The best experience for the user would be as @paulijar says "completely transparent so that the user wouldn't see any difference in browsing and playing tracks, regardless of if the tracks are stored as individual MP3 files or as part of some larger bundle described by a CUE file".

**Comment by coatmaker618** [comment]
@ei8fdb   exactly.  You will see
```
Album:
- album_name.wav
- album_name.cue
```

The wav (as it will likely be uncompressed by the program that extracts the audio from the CD--but could always be compressed either losslessly or lossy by the user later)  will be the entire length of the CD & the cue file will contain metadata like:
- track/song names
- track numbers
- start time (from which you can derive length)
- artist names
- etc.

Wikipedia has an example CUE file.
https://en.wikipedia.org/wiki/Cue_sheet_(computing)

**Comment by ei8fdb** [comment]
> As you can see, very simple but very powerful.

OK I understand. So example user experiece - the user:
- adds a folder with a wave and cue file to their Music folder
- Music app scans and finds it and adds it to the library
- user navigates to the album
- they see the list of tracks, each one separate
- user clicks on, e.g. "track 1" it plays

Like that?

**Comment by coatmaker618** [comment]
@ei8fdb exactly.  Clicking on track 1 would play it as if it was just that 1 track, not the start of the whole album/CD.

**Comment by ei8fdb** [comment]
> @ei8fdb exactly. Clicking on track 1 would play it as if it was just that 1 track, not the start of the whole album/CD.

Understand. So then it makes sense to me that the user experience for the person playing the music, should be the same as if they were playing an album with seperate files.
